changelog/@unreleased/pr-4287.v2.yml

```diff
@@ -0,0 +1,8 @@
+type: improvement
+improvement:
+  description: Improved tracking of the unreadable timestamp, and thus accuracy of
```

Comment:
nit: technically, it was not just falling behind due to the unreadable timestamp, it was because the punch table was not updated so when we would try to get the real time correlated with the last swept timestamp it would be stale.